### PR TITLE
Integrate new license checker package into image and package builds.

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] If you are adding/removing a .spec file that has multiple-versions supported, please add [@microsoft/cbl-mariner-multi-package-reviewers](https://github.com/orgs/microsoft/teams/cbl-mariner-multi-package-reviewers) team as reviewer [(Eg. golang has 2 versions 1.18, 1.21+)](https://github.com/microsoft/azurelinux/tree/2.0/SPECS/golang)
- [ ] Ready to merge

---

###### Summary 

The license checker added in https://github.com/microsoft/azurelinux/pull/9060 needs to be integrated into the day-to-day build process. Currently its configured to operate with warnings only, since there are a lot of packages with licensing issues.

This change adds two integrations:
- Package build
  - As each node in the graph is processed, pass the associated RPMs to an instance of the `LicenseChecker`. The rpms are copied into a working directory and the scan is run. Each scan returns intermediate data (yes/no for errors & warnings) while the actual results are accumulated in the `LicenseChecker`. Once the full graph has been built the results are saved to a .json file and a summary file. The build summary at the end of the build will also display each package that has errors/warnings. 
image

- Image build
  - Use the existing image target added in #9060 as a dependency of the image build process. It depends on the image package cache folder so it will be run after the fetcher runs, but before the image itself is created.

###### Change Log  

- Add `pkglicensechecker.go` to `schedulerutils`. It provides `PackageLicenseChecker` which makes a `LicenseChecker` suitable for use with the scheduler.
- Add build requirement `$(STATUS_FLAGS_DIR)/imager_disk_output.flag` -> `$(license_results_file_img)`

###### Does this affect the toolchain?  

**YES/NO**

###### Test Methodology

- Local testing

### CODE REVIEW

**Comment by dmcilvaney** [comment]
Leaving as draft for now, depends on https://github.com/microsoft/azurelinux/pull/9883 and https://github.com/microsoft/azurelinux/pull/9060.

File: toolkit/tools/scheduler/schedulerutils/pkglicensechecker.go [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+	// Clear the RPM directory before copying the new RPMs
+	err = file.RemoveDirectoryContents(p.rpmDirPath)
+	if err != nil {
+		err = fmt.Errorf("failed to clear RPM directory before scan:\n%w", err)
+		return false, false, err
+	}
+
+	// Copy the RPMs into the check directory
+	for _, rpmPath := range rpmPaths {
+		err = file.Copy(rpmPath, filepath.Join(p.rpmDirPath, filepath.Base(rpmPath)))
```
Comment (cwize1) [inline toolkit/tools/scheduler/schedulerutils/pkglicensechecker.go:124]: Why do you need to copy the RPMs? Seems wasteful.

Comment (cwize1) [inline toolkit/tools/scheduler/schedulerutils/pkglicensechecker.go:124]: Use symlink to avoid an actual copy?

**Review by cwize1 — APPROVED** [review]

File: toolkit/tools/scheduler/schedulerutils/pkglicensechecker.go (hunk 1 above)
Comment (dmcilvaney) [reply toolkit/tools/scheduler/schedulerutils/pkglicensechecker.go:124]: The license package was built to scan a directory. Maybe and advanced mode that can filter to specific files might be useful but this was quick and easy. I'll look at symlinks like you suggested, get the best of both worlds.

Comment (dmcilvaney) [reply toolkit/tools/scheduler/schedulerutils/pkglicensechecker.go]: Needed to use a hardlink, the symlink breaks when you enter the chroot.

**Pull request merged**